File: 백준/Silver/11047. 동전 0/동전 0.py
import sys

input = sys.stdin.read

data = input().strip().split()
N = int(data[0])
K = int(data[1])
coins = list(map(int, data[2:]))

answer = 0

for i in range(N - 1, -1, -1):
    answer += K // coins[i]
    K = K % coins[i]

print(answer)

File: 백준/Silver/11047. 동전 0/동전 0.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./ex.txt";
let input = fs.readFileSync(filePath).toString().trim().split("\n");
let [N, K] = input[0].split(" ").map(Number);
input.shift();
const coins = input.map(Number);
let answer = 0;

for (let i = N - 1; i >= 0; i--) {
  answer += Math.floor(K / coins[i]);
  K = K % coins[i];
}

console.log(answer);
